work_queue/src/work_queue.c

```diff
@@ -907,6 +935,20 @@ static int release_worker(struct work_queue *q, struct work_queue_worker *w)
 	return 1;
 }
 
+static int remove_fetch(struct work_queue *q, struct work_queue_worker *w) {
+	struct fetch *f;
+	list_first_item(q->ongoing_fetches);
+	while ((f=list_next_item(q->ongoing_fetches))) {
+		if (w == f->to_worker || w== f->from_worker) {
+			f->to_worker->ongoing_fetch = NULL;
+			f->from_worker->ongoing_fetch = NULL;
+			list_remove(q->ongoing_fetches, f);
+			return 1;
```

Comment:
memory leaks. Need to free(f) and perhaps whatever ->ongoing_fetch is pointing to.